packages/kbn-server-http-tools/src/get_listener.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import http from 'http';
+import https from 'https';
+import { getServerTLSOptions } from './get_tls_options';
+import type { IHttpConfig, ServerListener } from './types';
+
+interface GetServerListenerOptions {
+  configureTLS?: boolean;
+}
+
+export function getServerListener(
+  config: IHttpConfig,
+  options: GetServerListenerOptions = {}
+): ServerListener {
+  return configureHttp1Listener(config, options);
+}
+
+const configureHttp1Listener = (
+  config: IHttpConfig,
+  { configureTLS = true }: GetServerListenerOptions = {}
+): ServerListener => {
+  const useTLS = configureTLS && config.ssl.enabled;
+  const tlsOptions = useTLS ? getServerTLSOptions(config.ssl) : undefined;
+
+  const listener = useTLS
+    ? https.createServer({
+        ...tlsOptions,
+        keepAliveTimeout: config.keepaliveTimeout,
+      })
+    : http.createServer({
+        keepAliveTimeout: config.keepaliveTimeout,
+      });
+
+  listener.setTimeout(config.socketTimeout);
```

Comment:
We used to define a `.on('timeout', handler)`. Is this no longer needed?

Reply:
Na, good catch, I totally missed to re-add the

```ts
  server.listener.on('timeout', (socket) => {
    socket.destroy();
  });
```

TBH I'm not sure this is really useful (given I assume sockets will be destroyed on timeout anyway), but let's the purpose of the PR is not to test that, so I will add it back. Thanks!